source/adapters/opencl/device.cpp

```diff
@@ -1066,7 +1065,15 @@ UR_APIEXPORT ur_result_t UR_APICALL urDeviceGetInfo(ur_device_handle_t hDevice,
     return ReturnValue(UUID);
   }
 
-  case UR_DEVICE_INFO_KERNEL_SET_SPECIALIZATION_CONSTANTS: {
+  case UR_DEVICE_INFO_COMPONENT_DEVICES:
+  case UR_DEVICE_INFO_COMPOSITE_DEVICE:
+    // These two are exclusive to L0.
```

Comment:
just wondering if we have a number of things exclusive to a particular adapter if we should have something in the spec or generation stage to say its not needed for the other adapters, just skip it?

Reply:
Not really sure why I decided these aren't optional, looking at it in the spec and how it's used by sycl I think they should be. The big `UNSUPPORTED_ENUM` switch fall-through that we have in all the DeviceInfo implementations does look very automate-able although I'm not sure the relatively small maintenance burden justifies spending the time to be that clever about it

Reply:
yeah that makes sense